packages/flutter/test/material/ink_well_test.dart

```diff
@@ -1288,4 +1288,50 @@ void main() {
     await tester.pumpWidget(buildFrame(enabled: false));
     await tester.pumpAndSettle();
   });
+
+  testWidgets('InkWell does not attach semantics handler for onTap if it was not provided an onTap handler', (WidgetTester tester) async {
+    await tester.pumpWidget(Directionality(
+      textDirection: TextDirection.ltr,
+      child: Material(
+        child: Center(
+          child: InkWell(
+            onLongPress: () { },
+            child: const Text('Foo'),
+          ),
+        ),
+      ),
+    ));
+
+    expect(tester.getSemantics(find.bySemanticsLabel('Foo')), matchesSemantics(
+      label: 'Foo',
+      hasTapAction: false,
+      hasLongPressAction: true,
+      isFocusable: true,
+      textDirection: TextDirection.ltr,
+    ));
+
+    // Add tap handler back and confirm addition back to semantic actions.
+    await tester.pumpWidget(Directionality(
+      textDirection: TextDirection.ltr,
+      child: Material(
+        child: Center(
+          child: InkWell(
+            key: UniqueKey(),
```

Comment:
Seems like I shouldn't need this, but the semantics do not update correctly without it. Is this an issue with keepalive?

Reply:
KeepAlive shouldn't do anything in this widget tree. Why do you suspect it?

Reply:
If I do not add the unique key here, the semantics do not update

Reply:
Oh, I just found an embarrassing bug in semantics that's causing this. I'll post a PR to fix shortly.

Reply:
https://github.com/flutter/flutter/pull/69230

Reply:
You should be able to remove the unique key now.